cookbook/case_studies/feature_engineering/feast_integration/feast_wf.py

```diff
@@ -0,0 +1,351 @@
+    "horse_colic_stats:peripheral pulse",
+    "horse_colic_stats:surgical lesion",
+    "horse_colic_stats:abdominal distension",
+    "horse_colic_stats:nasogastric tube",
+    "horse_colic_stats:outcome",
+    "horse_colic_stats:packed cell volume",
+    "horse_colic_stats:nasogastric reflux PH",
+]
+DATABASE_URI = "https://cdn.discordapp.com/attachments/545481172399030272/861575373783040030/horse_colic.db.zip"
+DATA_CLASS = "surgical lesion"
+
+# %%
+# Next, we define the reference tasks. A :py:func:`flytekit.reference_task` references the Flyte tasks that have already been defined, serialized, and registered.
+# The primary advantage of using a reference task is to reduce the redundancy; we needn't define the task(s) again if we have multiple datasets that need to be feature-engineered.
+@reference_task(
+    project="flytesnacks",
+    domain="development",
+    name="feast_integration.feature_eng_tasks.mean_median_imputer",
+    version="v1",
+)
+def mean_median_imputer(
+    dataframe: pd.DataFrame,
+    imputation_method: str,
+) -> pd.DataFrame:
+    ...
+
+
+@reference_task(
+    project="flytesnacks",
+    domain="development",
+    name="feast_integration.feature_eng_tasks.univariate_selection",
+    version="v1",
+)
+def univariate_selection(
+    dataframe: pd.DataFrame, num_features: int, data_class: str, feature_view_name: str
+) -> pd.DataFrame:
+    ...
+
+
+# %%
+# .. note::
+#
+#   The ``version`` varies depending on the version assigned during the task registration process.
+
+# %%
+# We define a task to convert data frame to a parquet file. A parquet file is essential for Feast.
+@task
+def df_to_parquet(df: pd.DataFrame, column_name: str) -> FlyteFile:
+    df[column_name] = pd.to_datetime(df[column_name])
+    parquet_file = "horse_colic.parquet"
+    df.to_parquet(parquet_file)
+    return FlyteFile(parquet_file)
+
+
+# %%
+# We define two tasks, namely ``store_offline`` and ``retrieve_offline`` to store and retrieve the historial features.
+#
+# .. list-table:: Decoding the ``Feast`` Jargon
+#    :widths: 25 25
+#
+#    * - ``FeatureStore``
+#      - A FeatureStore object is used to define, create, and retrieve features.
+#    * - ``Entity``
+#      - Represents a collection of entities and associated metadata. It's usually the primary key of your data.
+#    * - ``FeatureView``
+#      - A FeatureView defines a logical grouping of serveable features.
+#    * - ``FileSource``
+#      - File data sources allow for the retrieval of historical feature values from files on disk for building training datasets, as well as for materializing features into an online store.
+#    * - ``fs.apply()``
+#      - Register objects to metadata store and update related infrastructure.
+#    * - ``get_historical_features()``
+#      - Enrich an entity dataframe with historical feature values for either training or batch scoring.
+@task
+def store_offline(parquet_file: FlyteFile, repo_path: str) -> (str, str):
```

Comment:
Can we just create it like a pre-existing task plugin? Where given a parquet file the task will automatically upload the data to Feast Offline feature store?

Reply:
@kumare3: What do we do about this? When I write a plugin, the input will not just be a parquet file, we'll have to take features, primary key, etc. Similarly, when retrieving the offline features, the user has to give the primary key and datetime values. The same applies to online features as well.